### PR TITLE
proto: update the bitstring dep

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This is an alternative version to [1] that
update only the necessary information around our codebase to upgrade the bitstring package.

This upgrade is necessary for the user of our package that will find conflics in their codebase about packages version.

[1] https://github.com/ElementsProject/lightning/pull/6585

### CODE REVIEW

File: contrib/pyln-proto/pyln/proto/zbase32.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import bitstring  # type: ignore
```
Comment (cdecker) [inline contrib/pyln-proto/pyln/proto/zbase32.py:1]: Does `bitstring` now come with type annotations?

Comment (dni) [reply contrib/pyln-proto/pyln/proto/zbase32.py:1]: i am using it here https://github.com/lnbits/bolt11 and neither mypy, pyright nor ruff are not complaining :).

Comment (dni) [reply contrib/pyln-proto/pyln/proto/zbase32.py:1]: i had type ignores all over the place in bolt11 before i updated to ^4

**Comment by nepet** [comment]
Rebased

File: contrib/pyln-proto/pyln/proto/zbase32.py (hunk 1 above)
Comment (nepet) [reply contrib/pyln-proto/pyln/proto/zbase32.py:1]: From the `bitsting` release notes: ` Type hints added throughout the code.` for `v4.0.1`

**Comment by nepet** [comment]
ACK 073584eee5a879c439bd8a4195357f01bbede72e

**Comment by vincenzopalazzo** [comment]
Thanks @nepet I was forgetting about this!

**Pull request merged**